Misc/NEWS.d/next/Library/2025-01-06-10-37-27.gh-issue-128384.V0xzwH.rst

```diff
@@ -0,0 +1,5 @@
+Add locking to :mod:`warnings` to avoid some data races when free-threading
+is used.  Change ``_warnings_runtime_state.mutex`` to be a recursive mutex
+and expose it to :mod:`warnings`, via the :func:`_acquire_lock` and
+:func:`_release_lock` functions.  Hold the lock when ``filters`` and
+``_filters_version`` are updated.
```

Comment:
```suggestion
Add locking to :mod:`warnings` to avoid some data races when free-threading
is used.  Change ``_warnings_runtime_state.mutex`` to be a recursive mutex
and expose it to :mod:`warnings`, via the :func:`!_acquire_lock` and
:func:`!_release_lock` functions. The lock is held when ``filters`` and
``_filters_version`` are updated.
```

This would fix the docs CI